polaris/polaris/utils.py

```diff
@@ -185,95 +148,39 @@ def create_stellar_deposit(transaction: Transaction) -> bool:
             return False
     # otherwise, create the envelope and sign it with the distribution account's secret
     else:
-        multisig = False
         distribution_acc, _ = get_account_obj(
             Keypair.from_public_key(transaction.asset.distribution_account)
         )
         envelope = create_transaction_envelope(transaction, distribution_acc)
         envelope.sign(transaction.asset.distribution_seed)
-        transaction.envelope_xdr = envelope.to_xdr()
 
     try:
-        return submit_stellar_deposit(transaction, multisig=multisig)
-    except RuntimeError as e:
-        transaction.status_message = str(e)
+        submit_stellar_deposit(transaction, envelope)
+    except (RuntimeError, BaseHorizonError) as e:
+        transaction.status_message = f"{e.__class__.__name__}: {e.message}"
         transaction.status = Transaction.STATUS.error
         transaction.save()
         logger.error(transaction.status_message)
         return False
-
-
-def handle_bad_signatures_error(e, transaction, multisig=False):
-    err_msg = (
-        f"Horizon returned a {e.__class__.__name__} on transaction "
-        f"{transaction.id} submission."
-    )
-    logger.error(err_msg)
-    logger.error(
-        f"Transaction {transaction.id} envelope XDR: {transaction.envelope_xdr}"
-    )
-    logger.error(f"Transaction {transaction.id} result XDR: {e.result_xdr}")
-    if multisig:
-        logger.error(
-            f"Resetting the Transaction.envelope_xdr for {transaction.id}, "
-            "updating Transaction.pending_signatures to True"
-        )
-        channel_account, _ = get_account_obj(
-            Keypair.from_public_key(transaction.channel_account)
-        )
-        transaction.envelope_xdr = create_transaction_envelope(
-            transaction, channel_account
-        ).to_xdr()
-        transaction.pending_signatures = True
-        transaction.status = Transaction.STATUS.pending_anchor
     else:
-        transaction.status = Transaction.STATUS.error
-        transaction.status_message = (
-            f"Horizon returned a {e.__class__.__name__} on transaction "
-            f"{transaction.id} submission"
-        )
-    transaction.save()
+        return True
```

Comment:
This else block is a bit confusing to me.
Can you confirm that this is reachable after we hit `submit_stellar_deposit(transaction, envelope)`?

Reply:
Yea, `try/catch` has an `else` clause which is executed if `catch` isn't. This differs from `finally` which executes regardless of `catch` or `else`.

Reply:
I learned something new in python today
I had no idea that this was valid syntax.
https://docs.python.org/3/tutorial/errors.html#handling-exceptions
```
for arg in sys.argv[1:]:
    if arg is Blah:
        print('BLAH', arg)
    else:
        print('NOT BLAH', arg)    
    try:
        f = open(arg, 'r')
    except OSError:
        print('cannot open', arg)
    else:
        print(arg, 'has', len(f.readlines()), 'lines')
        f.close()
```
Thanks for clearing that up!